Mage.Sets/src/mage/cards/q/QasaliAmbusher.java

```diff
@@ -2,23 +2,38 @@
 
 import java.util.UUID;
 import mage.MageInt;
-import mage.abilities.ActivatedAbilityImpl;
-import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.Ability;
+import mage.abilities.condition.AndCondition;
+import mage.abilities.condition.Condition;
+import mage.abilities.condition.common.AttackedThisStepCondition;
+import mage.abilities.condition.common.PermanentsOnTheBattlefieldCondition;
+import mage.abilities.costs.AlternativeCostSourceAbility;
+import mage.abilities.decorator.ConditionalAsThoughEffect;
 import mage.abilities.effects.common.continuous.CastAsThoughItHadFlashSourceEffect;
 import mage.abilities.keyword.ReachAbility;
 import mage.cards.CardImpl;
 import mage.cards.CardSetInfo;
 import mage.constants.*;
 import mage.filter.common.FilterControlledLandPermanent;
-import mage.game.Game;
-import mage.game.combat.CombatGroup;
+import mage.filter.predicate.Predicates;
+import mage.watchers.common.PlayerAttackedStepWatcher;
 
 /**
  *
  * @author Plopman
  */
 public final class QasaliAmbusher extends CardImpl {
 
+    private static final FilterControlledLandPermanent filter = new FilterControlledLandPermanent();
+
+    static {
+        filter.add(Predicates.and(SubType.FOREST.getPredicate(), SubType.PLAINS.getPredicate()));
```

Comment:
Filters works per permanent. So it will fails on any basic lands (it uses `and` predicate, e.g. one permanent must be forest and plains at the same time).